### DISCUSSION TITLE
Uncaught DOMException: Failed to execute 'btoa' on 'Window'

### DISCUSSION THREAD

**Discussion by v3ss0n** [Budibase/budibase#7847] [General]
`Uncaught DOMException: Failed to execute 'btoa' on 'Window': The string to be encoded contains characters outside of the Latin1 range.`

It seems that accours when stage is persisted and it incudes and emoji. I used a button to set emoji as state that used to filter table , after that all the tables linked rows cannot be clicked , failing with above error.

After clearing all site data , and removing the button it works again , but still having problem if unicode data is persisted .

SO Answer here : https://stackoverflow.com/questions/23223718/failed-to-execute-btoa-on-window-the-string-to-be-encoded-contains-characte

**Comment by melohagan** [comment]
Hey @v3ss0n 

I closed your issue: https://github.com/Budibase/budibase/issues/7848

Looks like you found it to be fixed? Did it prevent you from using the app builder / client apps in any way?

**Comment by v3ss0n** [comment]
Yes , if i persist`states` it breaks the app . Work around was i stopped using emojis inside states. 
So it can still break if emojis are persist in `states`

**Reply by melohagan** [reply]
Apologies, I've re-opened!

**Reply by v3ss0n** [reply]
Thanks  , i will try on latest.

**Comment by VenSlazer** [comment]
I'm also facing the `Uncaught DOMException: Failed to execute 'btoa' on 'Window': The string to be encoded contains characters outside of the Latin1 range.` error when I try to open a modal from a button. It worked before but suddenly stopped working.

What's even more strange is the fact, that this error only appears on Google Chrome on my MacBook Pro M1. Google Chrome on Windows is running fine, as well as all other browsers on either Mac or Windows.
